siff-dr/siff-header.go

```diff
@@ -36,20 +36,15 @@ func setSiffFields(packet *netfilter.NFPacket, flags layers.IPv4Flag, capabiliti
 
 	/* Modify the ip layer information */
 	// compute new IHL and length
-	var IHLchange uint8 = (*ipLayer).IHL
-
 	if (flags & 0x03) == IS_SIFF {
 		(*ipLayer).IHL = 7
 	} else if (flags & 0x03) == (IS_SIFF | CAPABILITY_UPDATE) {
-		(*ipLayer).IHL = 8
+		(*ipLayer).IHL = 9
 	}
 
 	/* change the total length by the change in IHL * 4 to convert from
 	   32-bit words to bytes */
-	IHLchange = (*ipLayer).IHL - IHLchange
-	if IHLchange != 0 {
-		(*ipLayer).Length = (*ipLayer).Length + uint16(IHLchange)*4
-	}
+	(*ipLayer).Length = uint16((*ipLayer).IHL) * 4
```

Comment:
Can't the packets still have data? It looks like you're just throwing out any data and only keeping the header